pinot-spi/src/main/java/org/apache/pinot/spi/filesystem/PinotFS.java

```diff
@@ -144,6 +143,22 @@ boolean exists(URI fileUri)
   long length(URI fileUri)
       throws IOException;
 
+  /**
+   * Lists all the files and directories at the location provided.
+   * Lists recursively if {@code inputFilePattern} starts with "glob:**".
+   * Throws IOException if this abstract pathname is not valid, or if an I/O error occurs.
+   * @param fileUri location of file
+   * @return an array of strings that contains file paths
+   * @throws IOException on IO failure. See specific implementation
+   */
+  default String[] listFilesWithPattern(URI fileUri, String inputFilePattern)
```

Comment:
let's not add the default implementation in `PinotFS` since this is a very generic interface.

I would suggest let's add this as a static method in `SegmentGenerationJobSpec` 
```
  public static String[] listMatchedFiles(PinotFS pinotFs, URI fileUri, String includePattern, String excludePattern) {
    // ...
  }
```

Reply:
+1, the default implementation should  be to throw an exception asking if you have permission from your mom doing this :) We have already fielded one security issue in this regard.

Also, the API comments should clearly indicate that this is a potential security hole, and its usage is unsafe.